storage/remote/otlptranslator/prometheusremotewrite/metrics_to_prw.go

```diff
@@ -114,12 +138,7 @@ func (c *PrometheusConverter) FromMetrics(ctx context.Context, md pmetric.Metric
 					continue
 				}
 
-				var promName string
-				if settings.AllowUTF8 {
-					promName = otlptranslator.BuildMetricName(metric, settings.Namespace, settings.AddMetricSuffixes)
-				} else {
-					promName = otlptranslator.BuildCompliantMetricName(metric, settings.Namespace, settings.AddMetricSuffixes)
-				}
+				promName := c.metricNamer.Build(otlptranslator.Metric{Name: metric.Name(), Unit: metric.Unit(), Type: otelTypeToTranslatorType(metric)})
```

Comment:
I think it would be nice to have a otlptranslator.NewMetric() convenience function to automate this common pattern.  `c.metricNamer.Build(otlptranslator.NewMetric(metric)`  (or, "FromPMetric"?)

Reply:
That's a really good idea :)